src/brogue/MainMenu.c

```diff
@@ -617,31 +617,98 @@ boolean dialogChooseFile(char *path, const char *suffix, const char *prompt) {
     }
 }
 
-void scumMonster(creature *monst) {
+void printSeedCatalogMonsters(boolean includeAll) {
+    creature *theMonster;
+    char name[500];
+
+    for (theMonster = monsters->nextCreature; theMonster != NULL; theMonster = theMonster->nextCreature) {
+        seedCatalogMonsterName(name, theMonster);
+        upperCase(name);
+        if (theMonster->bookkeepingFlags & MB_CAPTIVE || theMonster->creatureState == MONSTER_ALLY || includeAll) {
+            printf("        %s\n", name);
+        }
+    }
+
+    for (theMonster = dormantMonsters->nextCreature; theMonster != NULL; theMonster = theMonster->nextCreature) {
+        seedCatalogMonsterName(name, theMonster);
+        upperCase(name);
+        if (theMonster->bookkeepingFlags & MB_CAPTIVE || theMonster->creatureState == MONSTER_ALLY || includeAll) {
+            printf("        %s\n", name);
+        }
+    }
+}
+
+void printSeedCatalogMonsterItems() {
+    creature *theMonster;
+    char name[500];
+
+    for (theMonster = monsters->nextCreature; theMonster != NULL; theMonster = theMonster->nextCreature) {
+        if (theMonster->carriedItem != NULL && theMonster->carriedItem->category != GOLD) {
+                seedCatalogItemName(theMonster->carriedItem, name, theMonster);
+                printf("        %s\n", name);
+        }
+    }
+
+    for (theMonster = dormantMonsters->nextCreature; theMonster != NULL; theMonster = theMonster->nextCreature) {
+        if (theMonster->carriedItem != NULL && theMonster->carriedItem->category != GOLD) {
+            seedCatalogItemName(theMonster->carriedItem, name, theMonster);
+            printf("        %s\n", name);
+        }
+    }
+}
+
+void printSeedCatalogFloorItems() {
+    item *theItem;
+    int gold = 0;
+    short piles = 0;
     char buf[500];
-    if (monst->bookkeepingFlags & MB_CAPTIVE) {
-        monsterName(buf, monst, false);
-        upperCase(buf);
-        printf("\n        %s (captive)", buf);
-        if (monst->machineHome > 0) {
-            printf(" (vault %i)", monst->machineHome);
+
+    for (theItem = floorItems->nextItem; theItem != NULL; theItem = theItem->nextItem) {
+        if (theItem->category == GOLD){
+            piles++;
+            gold += theItem->quantity;
+        } else if (theItem->category == AMULET) {
+        } else {
+            seedCatalogItemName(theItem, buf, NULL);
+            printf("        %s\n", buf);
         }
-    } else if (monst->creatureState == MONSTER_ALLY) {
-        monsterName(buf, monst, false);
-        upperCase(buf);
-        printf("\n        %s (allied)", buf);
-        if (monst->machineHome) {
-            printf(" (vault %i)", monst->machineHome);
+    }
+    if (piles == 1 ){
+        printf("        %i gold pieces\n", gold);
+    }
+    if (piles > 1 ){
+        printf("        %i gold pieces (%i piles)\n", gold, piles);
+    }
+}
+
+void printSeedCatalogAltars(){
+    short i, j;
+    short c_altars[100];
```

Comment:
Perhaps this would be clearer as a `boolean` type using `true`/`false` (see Rogue.h). Also the loop initialiser below can be replaced with a [partial list](https://en.cppreference.com/w/c/language/array_initialization); everything unspecified is set to 0.

```suggestion
    boolean c_altars[100] = {false};
```

Reply:
ok yeah that makes sense

Reply:
done